io_ogre/ogre/ogre_import.py

```diff
@@ -0,0 +1,1585 @@
+                uvcoordset = []
+                for vertex in vb.getElementsByTagName('vertex'):
+                    uvcoords = []
+                    for vt in vertex.childNodes:
+                        if vt.localName == 'texcoord':
+                            u = float(vt.getAttributeNode('u').value)
+                            v = -float(vt.getAttributeNode('v').value)+1.0
+                            uvcoords.append([u, v])
+
+                    if len(uvcoords) > 0:
+                        uvcoordset.append(uvcoords)
+                vertexdata['uvsets'] = uvcoordset
+    
+    return vertexdata
+
+
+def xCollectMeshData(meshData, xmldoc, meshname, dirname):
+    logger.info("* Collecting mesh data...")
+    
+    # Global has_skeleton
+    faceslist = []
+    subMeshData = []
+    allObjs = []
+    isSharedGeometry = False
+    sharedGeom = []
+    hasSkeleton = 'boneIDs' in meshData
+
+    # Collect shared geometry
+    if(len(xmldoc.getElementsByTagName('sharedgeometry')) > 0):
+        isSharedGeometry = True
+        for subnodes in xmldoc.getElementsByTagName('sharedgeometry'):
+            meshData['sharedgeometry'] = xCollectVertexData(subnodes)
+
+        if hasSkeleton:
+            for subnodes in xmldoc.getElementsByTagName('boneassignments'):
+                meshData['sharedgeometry']['boneassignments'] = xCollectBoneAssignments(meshData, subnodes)
+
+    # Collect submeshes data
+    for submeshes in xmldoc.getElementsByTagName('submeshes'):
+        for submesh in submeshes.childNodes:
+            if submesh.localName == 'submesh':
+                materialOrg = str(submesh.getAttributeNode('material').value)
+                # To avoid Blender naming limit problems
+                material = GetValidBlenderName(materialOrg)
+                sm = {}
+                sm['material'] = material
+                sm['materialOrg'] = materialOrg
+                for subnodes in submesh.childNodes:
+                    if subnodes.localName == 'faces':
+                        facescount = int(subnodes.getAttributeNode('count').value)
+                        sm['faces'] = xCollectFaceData(subnodes)
+
+                        if len(xCollectFaceData(subnodes)) != facescount:
+                            logger.debug("FacesCount doesn't match!")
+                            break
+
+                    if (subnodes.localName == 'geometry'):
+                        vertexcount = int(subnodes.getAttributeNode('vertexcount').value)
+                        sm['geometry'] = xCollectVertexData(subnodes)
+
+                    if hasSkeleton and subnodes.localName == 'boneassignments' and isSharedGeometry==False:
+                        sm['geometry']['boneassignments'] = xCollectBoneAssignments(meshData, subnodes)
+
+                subMeshData.append(sm)
+
+    meshData['submeshes'] = subMeshData
+
+    return meshData
+
+
+def xCollectMaterialData(meshData, materialFiles, folder):
```

Comment:
we probably can delete this, given the other parser you implemented

Reply:
Done

Reply:
Also removed convertXML() whose functionality was moved to util.py